File: app/src/main/java/com/fastaccess/ui/adapter/ProfileOrgsAdapter.java
package com.fastaccess.ui.adapter;

import android.view.ViewGroup;

import com.fastaccess.data.dao.model.User;
import com.fastaccess.ui.adapter.viewholder.ProfileOrgsViewHolder;
import com.fastaccess.ui.widgets.recyclerview.BaseRecyclerAdapter;
import com.fastaccess.ui.widgets.recyclerview.BaseViewHolder;

/**
 * Created by Kosh on 21 May 2017, 12:30 PM
 */

public class ProfileOrgsAdapter extends BaseRecyclerAdapter<User, ProfileOrgsViewHolder, BaseViewHolder.OnItemClickListener<User>> {

    @Override protected ProfileOrgsViewHolder viewHolder(ViewGroup parent, int viewType) {
        return ProfileOrgsViewHolder.newInstance(parent);
    }

    @Override protected void onBindView(ProfileOrgsViewHolder holder, int position) {
        holder.bind(getItem(position));
    }
}

File: app/src/main/java/com/fastaccess/github/ui/adapter/ProfileOrgsAdapter.kt
package com.fastaccess.github.ui.adapter

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.fastaccess.data.persistence.models.UserOrganizationNodesModel
import com.fastaccess.github.ui.adapter.viewholder.ProfileOrgsViewHolder

/**
 * Created by Kosh on 2018-11-17.
 */
class ProfileOrgsAdapter(private val list: ArrayList<UserOrganizationNodesModel>) : RecyclerView.Adapter<ProfileOrgsViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProfileOrgsViewHolder = ProfileOrgsViewHolder(parent)
    override fun getItemCount(): Int = list.size
    override fun onBindViewHolder(holder: ProfileOrgsViewHolder, position: Int) = holder.bind(list[position])

    fun insertNew(list: List<UserOrganizationNodesModel>) {
        this.list.apply {
            clear()
            addAll(list)
            notifyDataSetChanged()
        }
    }
}